[e2e] Add tests for invalid OperatorSources

SamiSousa opened the pull request:
- Creates NoSetupTestGroup for tests that don't require any
  resources upfront
- Testsuite for invalid OperatorSource tests
- Check for OperatorSource with invalid URL
- Check for OperatorSource with invalid endpoint
- Check for OperatorSource with non-existent regsitry namespace

These OperatorSource tests each create an invalid OperatorSource
and check that the status reaches the appropriate failed state.

SamiSousa commented:
@aravindhp @kevinrizza @awgreene

aravindhp commented on test/testsuites/opsrcnonexistentregistrynamespacetest.go:
```diff
@@ -0,0 +1,88 @@
+	}
+
+	return nil
+}
```
Please add new line.

SamiSousa commented on test/testgroups/noresourcetests.go:
```diff
@@ -0,0 +1,15 @@
+	"github.com/operator-framework/operator-marketplace/test/testsuites"
+)
+
+// NoResourceTestGroup runs test suites that do not require any setup
```
Great point, I'll rename and explain that this test group is for testing invalid operatorsources

SamiSousa commented on test/testsuites/opsrcinvalidendpointtest.go:
```diff
@@ -0,0 +1,82 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func OpSrcWithInvalidEndpoint(t *testing.T) {
```
Addressed

SamiSousa commented:
@aravindhp Addressed your comments

SamiSousa commented:
@tkashem @kevinrizza Could you take a look? Thank you!

awgreene commented on test/testsuites/opsrcinvalidendpointtest.go:
```diff
@@ -0,0 +1,82 @@
+package testsuites
```
Should each of these be in their own test suite? All tests focus on invalid `operator sources` which I would argue should be included together in a `InvalidOperatorSourceTestSuite`.

SamiSousa replied:
Sounds good to me. I will put these tests into a single test-suite

awgreene replied:
These could even go in the `opsrc-creation-test-suite`.

SamiSousa replied:
I considered that, but I didn't need the operatorsource that is created as part of the TestGroup. Although, it doesn't seem as though the `opsrccreationtests.go` uses it either

awgreene replied:
>I considered that, but I didn't need the operatorsource that is created as part of the TestGroup. 
That's a good point 

>Although, it doesn't seem as though the opsrccreationtests.go uses it either

The test suite runs a series of tests that depend on the `OpSrc` being created before executing.

SamiSousa replied:
Ah, that's right. I think it still makes more sense to have the invalid tests be a separate testsuite since they don't use the same operatorsource and because they test a different use case.

awgreene commented on test/testgroups/invalidopsrctests.go:
```diff
@@ -0,0 +1,15 @@
+)
+
+// InvalidOpSrcTestGroup runs test suites for invalid operatorsources
+func InvalidOpSrcTestGroup(t *testing.T) {
```
`Test Groups` are typically used to create any resources that testsuites / tests in the group depend on. In this case, you've created a `test group` called `InvalidOpSrcTestGroup` - I think it makes more sense to create a `test group` meant for test suites that require no set up.

SamiSousa replied:
Following the conversation on Kevin's PR, I'll update my tests to follow this format.

SamiSousa commented:
@awgreene @kevinrizza @aravindhp 
Updated to put each test inside one testsuite for invalid OperatorSources, and renamed the TestGroup to `NoSetupTestGroup` based on the conversation here: https://github.com/operator-framework/operator-marketplace/pull/170#discussion_r277372037

SamiSousa commented:
Tests are failing because I needed to rebase on master, since many helper functions changed which arguments they take. Didn't realize this would affect my PR. Just pushed the rebased changes.

aravindhp commented:
/approve cancel

SamiSousa commented:
Thank you @aravindhp ! I addressed your comments, please take a look!

kevinrizza commented:
/lgtm

SamiSousa commented:
/retest

SamiSousa commented:
/test e2e-aws

SamiSousa commented:
/test e2e-aws-console-olm
